pkg/controllers/dynakube/injection/reconciler.go

```diff
@@ -75,7 +78,7 @@ func (r *reconciler) Reconcile(ctx context.Context) error {
 	// because the 2 injection type we have share the label that the webhook is listening to, we can only clean that label up if both are disabled
 	// but we should only clean-up the labels after everything else is cleaned up because the clean-up for the secrets depend on the label still being there
 	// but we have to do the mapping before everything when its necessary
-	if !r.dk.OneAgent().IsAppInjectionNeeded() && !r.dk.MetadataEnrichmentEnabled() {
+	if !r.dk.OneAgent().IsAppInjectionNeeded() && !r.dk.MetadataEnrichmentEnabled() && (!r.dk.FeatureDownloadViaJob() || r.dk.OneAgent().IsCSIAvailable()) {
```

Comment:
```suggestion
	if !r.dk.OneAgent().IsAppInjectionNeeded() && !r.dk.MetadataEnrichmentEnabled() {
```
I don't think you need those checks, `IsAppInjectionNeeded` already covers it 
- `if !r.dk.OneAgent().IsAppInjectionNeeded()` ==> no injection ==> no-bootstapper injection